# Tugas Besar I IF2200 Probabilitas dan Statistika
### Penarikan Kesimpulan dan Pengujian Hipotesis

Dibuat oleh: <br>
Dhanika Novlisariyanti (13521132) <br>
Brigita Tri Carolina (13521156) <br>

## Load Data

In [2]:
import numpy as np 
import pandas as pd 
import matplotlib as plt

dataset = pd.read_csv("anggur.csv", names = ["fixed acidity","volatile acidity","citric acid","residual sugar","chlorides","free sulfur dioxide","total sulfur dioxide","density","pH","sulphates","alcohol","quality"])
display(dataset)

fixed acidity  volatile acidity  citric acid      residual sugar  \
0     fixed acidity  volatile acidity  citric acid      residual sugar   
1               5.9            0.4451       0.1813   2.049401312441085   
2               8.4            0.5768       0.2099   3.109590324079827   
3              7.54            0.5918       0.3248   3.673744380694597   
4              5.39            0.4201       0.3131   3.371814589273552   
...             ...               ...          ...                 ...   
996            7.96            0.6046       0.2662  1.5920477965168494   
997            8.48             0.408       0.2227  0.6819554112395263   
998            6.11            0.4841        0.372   2.377267103044589   
999            7.76             0.359       0.3208   4.294486017029563   
1000           5.87            0.5214       0.1883  2.1794904865090774   

               chlorides  free sulfur dioxide  total sulfur dioxide  density  \
0              chlorides  free sulfur dioxide  total sulfur dioxide  density   
1      0.070574489612481    16.59381751499067                 42.27   0.9982   
2      0.101681448549234    22.55551936905905                 16.01    0.996   
3     0.0724155344849959    9.316866115881428                 35.52    0.999   
4     0.0727545223136335    18.21229953192828                 41.97   0.9945   
...                  ...                  ...                   ...      ...   
996   0.0575545908451193   14.892444517787114                 44.61   0.9975   
997   0.0516272427261961    23.54896535693792                 25.83   0.9972   
998   0.0428056970303753    21.62458515197077                 48.75   0.9928   
999   0.0982764438587818   12.746186033238509                 44.53   0.9952   
1000  0.0529232046058149    16.20386394601281                 24.37   0.9983   

        pH  sulphates  alcohol  quality  
0       pH  sulphates  alcohol  quality  
1     3.27       0.71     8.64        7  
2     3.35       0.57    10.03        8  
3     3.31       0.64     9.23        8  
4     3.34       0.55    14.07        9  
...    ...        ...      ...      ...  
996   3.35       0.54    10.41        8  
997   3.41       0.46     9.91        8  
998   3.23       0.55     9.94        7  
999    3.3       0.66     9.76        8  
1000  3.29        0.7    10.17        7  

[1001 rows x 12 columns]

In [3]:
dataset.isnull().sum()

fixed acidity           0
volatile acidity        0
citric acid             0
residual sugar          0
chlorides               0
free sulfur dioxide     0
total sulfur dioxide    0
density                 0
pH                      0
sulphates               0
alcohol                 0
quality                 0
dtype: int64

## 1. Deskripsi Statistika
Menulis deskripsi statistika (Descriptive Statistics) dari semua kolom pada data yang bersifat numerik, terdiri dari mean, median, modus, standar deviasi, variansi, range, nilai minimum, maksimum, kuartil, IQR, skewness dan kurtosis. Boleh juga ditambahkan deskripsi lain.

## 2. Visualisasi
Membuat Visualisasi plot distribusi, dalam bentuk histogram dan boxplot untuk setiap kolom numerik. Berikan uraian penjelasan kondisi setiap kolom berdasarkan kedua plot tersebut.


## 3. Penentuan Apakah Tiap Kolom Berdistribusi Normal atau Tidak
Menentukan setiap kolom numerik berdistribusi normal atau tidak. Gunakan normality test yang dikaitkan dengan histogram plot.

## 4. Tes Hipotesis Satu Sampel
Melakukan test hipotesis 1 sampel, <br>
a. Nilai rata-rata pH di atas 3.29? <br>
b. Nilai rata-rata Residual Sugar tidak sama dengan 2.50? <br>
c. Nilai rata-rata 150 baris pertama kolom sulphates bukan 0.65? <br>
d. Nilai rata-rata total sulfur dioxide di bawah 35? <br>
e. Proporsi nilai total Sulfat Dioxide yang lebih dari 40, adalah tidak sama dengan 50% ? <br>

## 5. Tes Hipotesis Dua Sampel
a. Data kolom fixed acidity dibagi 2 sama rata: bagian awal dan bagian akhir kolom. Benarkah rata-rata kedua bagian tersebut sama? <br>
b. Data kolom chlorides dibagi 2 sama rata: bagian awal dan bagian akhir kolom. Benarkah rata-rata bagian awal lebih besar daripada bagian akhir sebesar 0.001? <br>
c. Benarkah rata-rata sampel 25 baris pertama kolom Volatile Acidity sama dengan rata-rata 25 baris pertama kolom Sulphates? <br>
d. Bagian awal kolom residual sugar memiliki variansi yang sama dengan bagian akhirnya? <br>
e. Proporsi nilai setengah bagian awal alcohol yang lebih dari 7, adalah lebih besar daripada, proporsi nilai yang sama di setengah bagian akhir alcohol? <br>